--- a/content/browser/render_process_host_impl.cc	2018-07-04 13:41:01.923430801 +1200
+++ b/content/browser/render_process_host_impl.cc	2018-07-04 13:40:49.151375681 +1200
@@ -3065,6 +3065,10 @@
     ServiceManagerConnection* service_manager_connection =
         BrowserContext::GetServiceManagerConnectionFor(browser_context_);
     connection_filter_controller_->DisableFilter();
+    /* XXX brendonj - service_manager_connection is often null */
+    if ( service_manager_connection ) {
+        service_manager_connection->RemoveConnectionFilter(connection_filter_id_);
+    }
     connection_filter_id_ =
         ServiceManagerConnection::kInvalidConnectionFilterId;
   }
