--- a/src/tests/python/Makefile.am	2015-08-24 16:48:59.000000000 +1200
+++ b/src/tests/python/Makefile.am	2015-08-24 19:07:47.953931377 +1200
@@ -7,8 +7,7 @@
 install-exec-local:
 	mkdir -p $(DESTDIR)/$(pythondir)/
 	PYTHONPATH=$(DESTDIR)/$(pythondir)/ \
-	    python setup.py install --prefix=$(DESTDIR)/$(prefix) \
-            $(DEB_PYTHON_INSTALL_ARGS)
+	    python setup.py install --prefix=$(DESTDIR)/$(prefix)
 
 clean-local:
 	python setup.py clean --all

--- a/src/tests/python/Makefile.in	2015-08-24 16:49:07.000000000 +1200
+++ b/src/tests/python/Makefile.in	2015-08-24 19:05:20.249728192 +1200
@@ -374,8 +374,7 @@
 install-exec-local:
 	mkdir -p $(DESTDIR)/$(pythondir)/
 	PYTHONPATH=$(DESTDIR)/$(pythondir)/ \
-	    python setup.py install --prefix=$(DESTDIR)/$(prefix) \
-            $(DEB_PYTHON_INSTALL_ARGS)
+	    python setup.py install --prefix=$(DESTDIR)/$(prefix)
 
 clean-local:
 	python setup.py clean --all
